consensus/core/src/dag_state.rs

```diff
@@ -998,81 +998,66 @@ impl DagState {
         commit_round.saturating_sub(self.context.protocol_config.gc_depth())
     }
 
+    /// Flushes unpersisted blocks, commits and commit info to storage.
+    ///
+    /// REQUIRED: when buffering a block, all of its ancestors and the latest commit which sets the GC round
+    /// must also be buffered.
+    /// REQUIRED: when buffering a commit, all of its included blocks and the previous commits must also be buffered.
+    /// REQUIRED: when flushing, all of the buffered blocks and commits must be flushed together to ensure consistency.
+    ///
     /// After each flush, DagState becomes persisted in storage and it expected to recover
     /// all internal states from storage after restarts.
-    pub(crate) fn flush(&mut self, finalized_commit_index: CommitIndex) {
-        self.flush_internal(finalized_commit_index, /* ensure_exists */ true);
+    pub(crate) fn flush(&mut self) {
+        self.flush_internal();
     }
 
     /// Flush all buffered commits to storage.
     #[cfg(test)]
     pub(crate) fn flush_all_in_test(&mut self) {
```

Comment:
No longer needed?

Reply:
Yeah it was kept in case specializing `flush()` for testing is needed again. But removing this for now.